Pdf4QtLib/sources/pdffont.cpp

```diff
@@ -96,14 +97,13 @@ const PDFSystemFontInfoStorage* PDFSystemFontInfoStorage::getInstance()
 QByteArray PDFSystemFontInfoStorage::loadFont(const FontDescriptor* descriptor, StandardFontType standardFontType, PDFRenderErrorReporter* reporter) const
```

Comment:
I think, special font handling for linux will be needed. There is implementation only for MS Windows. I think for current implementation, only embedded fonts will work.

Reply:
on the TODO list

Reply:
Why not using QFont?

Reply:
Unfortunately, this was not possible, because QFont doesn't have required interface (pdf handles fonts in a special way), even QRawFont does not suffice.

Reply:
I guess you have some simple pdf files that you use for testing without embedded fonts. Could I use them to test my font code?

Reply:
Sent via email.

Reply:
I spent some time looking at your code. What I understand is that you implemented an heuristic to find closest font possible using all the local fonts.
I started looking into a way to implement the exact same functionality using fontconfig. I think it's possible.
But finding a font with specific requirements is exactly the purpose of fontconfig.
It's up to you to decide but I see 2 solutions here.
Let me know what you think.

Reply:
Hello, yes, it is a heuristic to find closest font possible, using all local fonts. The reason, why it is implemented this way, is that Windows doesn't have same functionality as fontconfig. If it is possible to implement it via fontconfig, then please use fontconfig. It is better to use native linux approach, than implement it own way. Also, user can configure fontconfig. Thank you.

Reply:
Don't know if you speak French (maybe with Google translate) but you should look at this page:
https://linuxfr.org/users/pied/journaux/pdf-mais-que-fait-la-police
I also add a look at how poppler generates search patterns for fontconfig using the pdf fontdescriptor.
Let me know if this is the kind of behaviour you expect.
Thanks

Reply:
Sorry, I do not speak french. I think you can be inspired by it, but please do not copy code from poppler.